Sources/NIOHPACK/HuffmanCoding.swift

```diff
@@ -0,0 +1,243 @@
+        let written = self.setHuffmanEncoded(bytes: stringBytes)
+        self.moveWriterIndex(forwardBy: written)
+        return written
+    }
+    
+    fileprivate static func appendSym_fast(_ sym: HuffmanTableEntry, _ state: inout _EncoderState, bytes: UnsafeMutableRawBufferPointer) {
+        // will it fit as-is?
+        if sym.nbits == state.remainingBits {
+            bytes[state.offset] |= UInt8(sym.bits)
+            state.offset += 1
+            state.remainingBits = state.offset == bytes.count ? 0 : 8
+        } else if sym.nbits < state.remainingBits {
+            let diff = state.remainingBits - sym.nbits
+            bytes[state.offset] |= UInt8(sym.bits << diff)
+            state.remainingBits -= sym.nbits
+        } else {
+            var (code, nbits) = sym
+            
+            nbits -= state.remainingBits
+            bytes[state.offset] |= UInt8(code >> nbits)
+            state.offset += 1
+            
+            if nbits & 0x7 != 0 {
+                // align code to MSB
+                code <<= 8 - (nbits & 0x7)
+            }
+            
+            // we can short-circuit if less than 8 bits are remaining
+            if nbits < 8 {
+                bytes[state.offset] = UInt8(truncatingIfNeeded: code)
+                state.remainingBits = 8 - nbits
+                return
+            }
+            
+            // longer path for larger amounts
+            switch nbits {
+            case _ where nbits > 24:
+                bytes[state.offset] = UInt8(truncatingIfNeeded: code >> 24)
+                nbits -= 8
+                state.offset += 1
+                fallthrough
+            case _ where nbits > 16:
+                bytes[state.offset] = UInt8(truncatingIfNeeded: code >> 16)
+                nbits -= 8
+                state.offset += 1
+                fallthrough
+            case _ where nbits > 8:
+                bytes[state.offset] = UInt8(truncatingIfNeeded: code >> 8)
+                nbits -= 8
+                state.offset += 1
+            default:
+                break
+            }
+            
+            if nbits == 8 {
+                bytes[state.offset] = UInt8(truncatingIfNeeded: code)
+                state.offset += 1
+                state.remainingBits = state.offset == bytes.count ? 0 : 8
+            } else {
+                state.remainingBits = 8 - nbits
+                bytes[state.offset] = UInt8(truncatingIfNeeded: code)
+            }
+        }
+    }
+    
+    fileprivate mutating func ensureBitsAvailable(_ bits: Int) {
+        let bytesNeeded = bits / 8
+        if bytesNeeded <= self.writableBytes {
+            // just zero the requested number of bytes before we start OR-ing in our values
+            self.withUnsafeMutableWritableBytes { ptr in
+                ptr.baseAddress!.assumingMemoryBound(to: UInt8.self).assign(repeating: 0, count: bytesNeeded)
```

Comment:
I appreciate that we only recently changed our rules here but we should not longer blindly assume the memory is bound to `UInt8`. To just zero out the memory I'd suggest

    ptr.copyBytes(from: repeatElement(0, count: bytesNeeded))